winter/schema/type_inspection.py

```diff
@@ -0,0 +1,195 @@
+import collections
+import datetime
+import decimal
+import enum
+import inspect
+import typing
+import uuid
+from collections import OrderedDict
+
+import dataclasses
+from drf_yasg import openapi
+from rest_framework.settings import api_settings as rest_settings
+
+from winter.type_utils import NoneType
+from winter.type_utils import UnionType
+from winter.type_utils import get_origin_type
+from winter.type_utils import is_optional
+
+TYPE_DECIMAL = openapi.TYPE_STRING if rest_settings.COERCE_DECIMAL_TO_STRING else openapi.TYPE_NUMBER
+TYPE_NONE = 'none'
+_resolvers_by_type: typing.Dict[typing.Type, typing.Tuple[typing.Callable, typing.Optional[typing.Callable]]] = {}
+
+
+def register(*types: typing.Tuple[typing.Type], checker=None):
```

Comment:
Isn't the name is too general? How external usage will look like?

```
from winter.schema.type_inspection import register

@register
def inspect_custom_type(...)
```

or 

```
import winter.schema.type_inspection
@winter.schema.type_inspection.register
def inspect_custom_type(...)
```

May be rename it to `register_type_inspector`?
```
import winter.schema

@winter.schema.register_type_inspector
def inspect_custom_type(...)
```

As a bonus, it will be consistent with `register_encoder`.